src/main/java/org/verdictdb/coordinator/VerdictSingleResult.java

```diff
@@ -16,147 +16,36 @@
 
 package org.verdictdb.coordinator;
 
-import com.google.common.base.Optional;
-import com.rits.cloning.Cloner;
 import org.verdictdb.commons.AttributeValueRetrievalHelper;
-import org.verdictdb.connection.DbmsQueryResult;
 import org.verdictdb.connection.DbmsQueryResultMetaData;
 
 /**
  * Represents the result set returned from VerdictDB to the end user.
  *
  * @author Yongjoo Park
  */
-public class VerdictSingleResult extends AttributeValueRetrievalHelper {
+public abstract class VerdictSingleResult extends AttributeValueRetrievalHelper {
```

Comment:
I feel like interface makes more sense here too. We can place `extends AttributeValueRetrievalHelper` in individual classes. Let me know what you think.

Reply:
If we change VerdictSingleResult to interface, is it safe to change VerdictResultSet  L32 to VerdictResultSetFromDbmsQueryResult? 

```
public class VerdictResultSet implements ResultSet {

  private VerdictSingleResult queryResult;
```

Reply:
What would be the problem if we keep VerdictSingleResult there?

Reply:
I think it must be VerdictSingleResult to cover both the one based on DbmsQueryResult and the one based on ListData.

Reply:
VerdictResultSet is using VerdictSingleResult to implement get method such as getArray(). So it suppose that VerdictSingleResult  extends AttributeValueRetrievalHelper.

Reply:
I see. Let's do this.

```java
public interface VerdictSingleResult { 
  (copy those empty get interface methods from DbmsQueryResult [1])
}

public class VerdictSingleResultFromDbmsQueryResult 
    extends AttributeValueRetrievalHelper { ... }

public class VerdictSingleResultFromListData 
    extends AttributeValueRetrievalHelper { ... }
```

[1] https://github.com/mozafari/verdictdb/blob/master/src/main/java/org/verdictdb/connection/DbmsQueryResult.java

Reply:
Does that make no relationship between VerdictSingleResult and its subclasses?

Reply:
Oh. It should have been like this:
```java
public class VerdictSingleResultFromDbmsQueryResult 
    implements VerdictSingleResult
    extends AttributeValueRetrievalHelper { ... }

public class VerdictSingleResultFromListData 
    implements VerdictSingleResult
    extends AttributeValueRetrievalHelper { ... }
```